src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -175,19 +171,15 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
         /// }).ConfigureAwait(continueOnCapturedContext: false);                
         /// </code>
         /// </example>
-        public Task<int> ExecuteAsync(string query, Action<IDbCommand> parameterProvider, CancellationToken cancellationToken = default(CancellationToken), CommandBehavior commandBehavior = CommandBehavior.Default)
+        public async Task<int> ExecuteAsync(string query, Action<IDbCommand> parameterProvider, CancellationToken cancellationToken = default(CancellationToken), CommandBehavior commandBehavior = CommandBehavior.Default)
         {
             //If the query is something else that is not acceptable (e.g. an empty string), there will an appropriate database exception.
             if(query == null)
             {
                 throw new ArgumentNullException("query");
```

Comment:
I think that it'd be better to have a `TaskCompletionSource` and use `SetException`.